### ISSUE TITLE
Implement support for structured dimensions in metadata editor

### ISSUE THREAD

**Issue by rla2118** [DCLP/dclpxsltbox#192]
We need to allow for dimension values in the metadata (height, width, and thickness), both for single objects/fragments and for multiple.

**Comment by paregorios** [comment]
I am currently researching this.

**Comment by paregorios** [comment]
The *EpiDoc Guidelines* follow the *TEI Guidelines* in providing [guidance for the markup of dimensions](http://www.stoa.org/epidoc/gl/latest/supp-description.html) like height, width, and thickness and does so with respect to the physical text-bearing object, the "campus" (writing field), and letters. 

I've done some spelunking in idp.data and I find that none of the collections make use of this facility at present. APIS does include dimensions, but these are simply textual and are included with other undifferentiated information in the support description. E.g.: [columbia.apis.3](http://papyri.info/apis/columbia.apis.3):

```xml
<physDesc>
    <objectDesc>
        <supportDesc>
            <support>1 ostracon ; tan clay ; 13.5 x 7.5 cm</support>
        </supportDesc>
    </objectDesc>
</physDesc>
```

EpiDoc/TEI would say that, if you want to markup the dimensions, you should add additional tags, like this:

```xml
<physDesc>
    <objectDesc>
        <supportDesc>
            <support>
                <ab>1 <objectType>ostracon</objectType> ; <material>tan clay</material> ;
                     <dimensions unit="cm">
                         <width>13.5</width> x 
                         <height>7.5 cm</height></dimensions></ab>
            </support>
        </supportDesc>
    </objectDesc>
</physDesc>
```

Before we can move to implementation, we need to discuss the use case. Do we merely want the ability to store a descriptive string about the support -- meant for human readers -- that might include dimensions? Or do we want the ability (or to facilitate for others the ability) to do computing on the dimensions? If the former, the editor will be modified to support text entry for a "support description" field, if it does not already cater for such.  If the latter, then a more sophisticated set of affordances will have to be added to the editor. Either way, some modification to the XSLT will be required, but I don't think it will be significant. 

@rla2118 how would you like to proceed?

**Comment by paregorios** [comment]
I note now that dimensions marked up were specified in the February 2016 "editor mask" example file. viz #173

**Comment by paregorios** [comment]
@Edelweiss has editor support been implemented for dimensions as described above?

**Comment by paregorios** [comment]
cc @HolgerEssler @jcowey

**Comment by Edelweiss** [comment]
material
```
/TEI/teiHeader/fileDesc/sourceDesc/msDesc/physDesc/objectDesc/supportDesc/support/material
```
width
```
/TEI/teiHeader/fileDesc/sourceDesc/msDesc/physDesc/objectDesc/supportDesc/support/dimensions/width[@unit='cm']
```
height
```
/TEI/teiHeader/fileDesc/sourceDesc/msDesc/physDesc/objectDesc/supportDesc/support/dimensions/height[@unit='cm']
```
pagination
```
/TEI/teiHeader/fileDesc/sourceDesc/msDesc/physDesc/objectDesc/supportDesc/foliation/dim[@type='pagination']
```

Add two text fields to SoSOL’s DCLP metadata editor under “Physical Description”.

**Comment by Edelweiss** [comment]
ok, [github](https://github.com/DCLP/sosol/commit/f41677cac848576ac3f8054569e263558ff74c0e), will soon be ready for testing on our development server (sometime today).

**Comment by jcowey** [comment]
Still to be tested

**Comment by jcowey** [comment]
Tested and works fine. Is not displayed.

**Comment by paregorios** [comment]
@jcowey what do you mean by "is not displayed." Is that a good thing or a bad thing?

**Comment by jcowey** [comment]
@paregorios "is not displayed" = when one clicks on preview, what is shown does not surface the entered data on width and height.
This will, I am almost certain, involve a XSLT tweak. I would imagine that the present XSLT would also not display width and height in your output files.
I am assuming that @rla2118 and @HolgerEssler  want to be able to see this data in HTML and in preview.

**Comment by paregorios** [comment]
@jcowey so, we probably need a new ticket for review and modification as necessary in the HTML view, yes?

**Comment by jcowey** [comment]
Yes that sounds right to me.

**Comment by paregorios** [comment]
I have now created issue #225 to address the XSLT concerns.
